pkg/provider/kubernetes/ingress/client.go

```diff
@@ -275,8 +297,53 @@ func extensionsToNetworking(ing marshaler) (*networkingv1beta1.Ingress, error) {
 	return ni, nil
 }
 
+func addServiceFromV1Beta1(ing *networkingv1.Ingress, old networkingv1beta1.Ingress) {
+	if old.Spec.Backend != nil {
+		port := networkingv1.ServiceBackendPort{}
+		if old.Spec.Backend.ServicePort.Type == intstr.Int {
```

Comment:
ServicePort is optional, we should test if it has the default value

Reply:
done :)